### ISSUE TITLE
Views image handling (with Backdrop image styles)

### ISSUE THREAD

**Issue by laryn** [civicrm/civicrm-backdrop#60]
The views image handling (using image styles) doesn't quite work for me out of the box. There are two issues here, I think:

1. The core contact image functionality doesn't work with image styles (as they should).
2. Contact images that were uploaded via `webform_civicrm` doubly don't work (they were probably never supported officially).

---

PR by @laryn: https://github.com/civicrm/civicrm-backdrop/pull/61

**Comment by laryn** [comment]
@herbdool Here's a PR that seems to work for me with CiviCRM core uploaded contact images as well as webform contact image uploads (both public and private).
https://github.com/civicrm/civicrm-backdrop/pull/61

**Comment by laryn** [comment]
@herbdool I made a slight tweak here. I am working on a new simple little directory on a site and was having an issue with core uploaded images -- I had the `return` one level to deep so it was returning nothing on core images.

**Comment by laryn** [comment]
If anyone else is able to test this one, I'd love to get it merged. (Selfishly... because I have a few sites that I keep patching after each CiviCRM update.)

**Comment by laryn** [comment]
@herbdool Any chance you will have time to review this one? It's been updated since you first looked at it.

**Comment by herbdool** [comment]
@laryn I'll try to take a look within the next few days

**Comment by herbdool** [comment]
@laryn in my testing it seems to do the job. I added some images to contacts via CiviCRM UI. Before applying the patch no contact images appear in a View. Then with the patch the contact images appear. Though they have an odd path, e.g.: `http://DOMAIN.ORG/files/styles/thumbnail/public/private/civicrm/custom/btsw2020_807989a86c05038c76beb210895f8638.png`. Odd in that it says `/public/private` but that might just be because I'm testing it on Pantheon.

Then I installed webform_civicrm, created a webform to upload images. Interesting it already shows the image I uploaded via Civi. I replace the image and it works.

I didn't test webform images in a private directory. I'm not sure best way to do that.

I also noticed that `$image['path']` got changed to `$image['uri']`. I assume the former doesn't get picked up by Backdrop?

Overall, it seems like the patch improves things.
